fvm/evm/types/block.go

```diff
@@ -73,13 +77,21 @@ func NewBlock(
 		TotalSupply:         totalSupply,
 		ReceiptRoot:         gethTypes.EmptyReceiptsHash,
 		TransactionHashRoot: gethTypes.EmptyRootHash,
+		PrevRandao:          prevRandao,
 	}
 }
 
 // NewBlockFromBytes constructs a new block from encoded data
 func NewBlockFromBytes(encoded []byte) (*Block, error) {
 	res := &Block{}
-	return res, gethRLP.DecodeBytes(encoded, res)
+	err := gethRLP.DecodeBytes(encoded, res)
+	if err != nil {
```

Comment:
Is it worth checking that the error is of the correct type?

Reply:
I tried that initially but simply all errors are related to the input not matching the output and non-fatal.